CMakeLists.txt

```diff
@@ -81,6 +81,13 @@ check_cxx_compiler_flag("--cuda-host-only -x hip" HAS_HIP)
 if(HAS_HIP)
     message(STATUS "Enable gpu backend")
     set(MIGRAPHX_ENABLE_GPU On CACHE BOOL "")
+    execute_process(COMMAND bash -c "${CMAKE_INSTALL_PREFIX}/bin/rocminfo | grep -o -m1 'gfx.*'"  OUTPUT_STRIP_TRAILING_WHITESPACE OUTPUT_VARIABLE GPU_TARGET)
+    set(GPU_TARGETS ${GPU_TARGET} CACHE STRING "")
+    if(GPU_TARGET)
+        message(STATUS "GPU_TARGETS set to: ${GPU_TARGETS}")
+    else()
+        message(FATAL_ERROR "No GFX target found via rocminfo.\n Please specify gfx target with -DGPU_TARGETS=<target>")
```

Comment:
This shouldn't be added to the cmake. It should be added to the Jenkinsfile by passing `-DGPU_TARGETS=$(op/rocm/bin/rocminfo | grep -o -m1 'gfx.*')`.

Reply:
Then what about local checks? Make check would fail when MIGRAPHX_ENABLE_GPU=ON since our make check would look at TARGET_GPU for a gfx value

Reply:
Ah so we're splitting this into #2234 then?

Reply:
> Then what about local checks?

We should pass `-DGPU_TARGETS` to cmake in jenkins so the local checks pass. 

> Make check would fail when MIGRAPHX_ENABLE_GPU=ON since our make check would look at TARGET_GPU for a gfx value

It does tell the user a workaround, but I guess its a little late. In #2234, I added a cmake error for when hip is broken, to let the user know at cmake time the workaround needed(ie `-DGPU_TARGETS`).

Reply:
Removed this all now and pushed things into Jenkinsfile